packages/request-manager/src/httpPool.ts

```diff
@@ -0,0 +1,74 @@
+import http from 'http';
+import { getWeakRandomId } from '@trezor/utils';
+import { InterceptorOptions } from './types';
+
+interface RequestData {
+    id: string;
+    requestTime: number;
+    host: string;
+}
+
+export class RequestPool {
+    requestPool: RequestData[] = [];
+    requestTimeoutLimit = 1000 * 10;
+    isNetworkMisbehaving = false;
+    interceptorOptions: InterceptorOptions;
+
+    constructor(interceptorOptions: InterceptorOptions) {
+        this.interceptorOptions = interceptorOptions;
+    }
+
+    removeRequest(id: string) {
+        this.requestPool = this.requestPool.filter(req => req.id !== id);
+    }
+
+    addRequest(request: http.ClientRequest) {
+        const id = getWeakRandomId(10);
+        const { host } = request;
+        this.requestPool.push({
+            id,
+            requestTime: new Date().getTime(),
+            host,
+        });
+        request.on('response', (response: http.IncomingMessage) => {
+            const requestFromPool = this.requestPool.find(req => req.id === id);
+            if (requestFromPool) {
+                const timeRequestTook = new Date().getTime() - requestFromPool.requestTime;
+                const { statusCode } = response;
+
+                this.isNetworkMisbehaving =
+                    timeRequestTook > this.requestTimeoutLimit || !statusCode || statusCode >= 400;
```

Comment:
My idea with network misbehaving is to expand it and make it more rich in the way it handles different situations so I wanted to start by notifying error when whatever http error related code.
But since it does not make anything right now I agree let's simplify it bb381d81ec42ea3577c7dbc2625170e6db4e554f

Reply:
I think it's better this way since it allows to distinguish between network problems, causing responses taking more time and all other errors :)